evennia/contrib/actions/README.md

```diff
@@ -0,0 +1,372 @@
+
+Movement between rooms is based on two actions, MoveOut and MoveIn. Thus, to move between two rooms A and B, characters must first move out of room A towards the exit to room B, and then into room B from the exit back to room A (if any such exit exists). It takes time to both enter a room and to leave it, as determined by the character's movement speed and the "distance" attribute of the rooms' exits. Moving out of a room cancels any actions the character is performing on targets within the room, as well as any actions being performed on the character by other characters within the room.
+
+For movement to work properly, there must be at most one exit from a given room A to another room B. If this is not the case, the distance used to measure the time required to move into A from B will be based on the distance value of only one of these exits. Furthermore, if there is no exit back from room A to room B at all, the time required to move into A from B will be the same as the time required to move out of B towards A.
+
+## Hooking up the action system
+
+The action contrib stores its files in its own python package. For convenience, all the classes and functions that the end-user will need are referenced in the package's \_\_init\_\_.py file, so that one can import them directly:
+
+> from evennia.contrib.actions import *<feature>*
+
+To use the action system, the characters, rooms and exits in your game must be subclassed from ActionCharacter, ActionRoom and ActionExit respectively. Simply put the following lines in the appropriate files:
+
+> from evennia.contrib.actions import ActionCharacter
+
+> from evennia.contrib.actions import ActionRoom
+
+> from evennia.contrib.actions import ActionExit
+
+And then subclass the appropriate [typeclasses](https://github.com/evennia/evennia/wiki/Typeclasses "typeclasses").
+Each of the ActionCharacter and ActionRoom objects has an attribute called "actions" that stores all methods and properties related to the action system. When the object is initialized, its actions.setup() method runs, loading the default parameters for the object. Likewise, running setup() from the evennia.contrib.actions modulecalls the actions.setup() methods for all objects that are subclassed from either ActionCharacter or ActionRoom.
+
+In any subclass you make for the ActionCharacter and ActionRoom classes, you must be sure to fill in several members of the "actions" attribute for the object being created, as well as call [super](https://docs.python.org/2/library/functions.html#super "the super function"), within at_object_creation. 
+
+For subclasses of __ActionCharacter__, you should do:
+
+> at_object_creation(self):
+>     self.actions.movespeed = <your function here>
```

Comment:
I was using the term "attribute" to mean a barebones python attribute, the way y is an attribute of x.y . I will call it a "handler" instead to avoid confusion.

The super() call will call self.actions.setup() without the override flag, meaning that if the custom-set properties have already been set, they will not be set again. It's not an issue.